protos/flyteidl/core/security.proto

```diff
@@ -42,7 +42,27 @@ message Secret {
     // if the underlying key management system cannot satisfy that requirement. If not provided, the default location
     // will depend on the key management system.
     // +optional
-    MountType mount_requirement = 4;
+    MountType mount_requirement = 4 [deprecated=true];
+
+    // mount_target is optional. Indicates how the secret should be mounted. Secrets can be mounted as file, or as 
+    // environment variable. 
+    // +optional
+    oneof mount_target {
+        MountEnvVar env_var = 5;
+        MountFile file = 6;
```

Comment:
since the existing `MountType` enum was only used in the `mount_requirement` field, it'd be effectively deprecated as well right?  Can we add the flag there as well?

Also since the existing enum had an Any concept, should we add that as well to the one of?

Reply:
Nm - let's ignore this comment